Refine PULL_REQUEST_TEMPLATE.md (#13727)
JabRef/jabref: CONTRIBUTING.md

1. Follow the steps at [Pre Condition 3: Code on the local machine](https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/pre-03-code) to a) create a fork and b) have the fork checked out on your local machine
2. Ensure that you followed the [steps to set up a local workspace](https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/) to have the code running properly in IntelliJ.

> 3. Read our [architecture and components](https://devdocs.jabref.org/architecture-and-components).

Reviewer: ```suggestion
3. Read about our [architecture and components](https://devdocs.jabref.org/architecture-and-components).
```

4. **Create a new branch** (such as `fix-for-issue-121`). Be sure to create a **separate branch** for each improvement you implement.
5. Refer to our [code how-tos](https://devdocs.jabref.org/code-howtos) if you have questions about your implementation.
6. Implement and test your changes.
   * Create tests for your changes. Maybe even use [Test-driven Development](https://en.wikipedia.org/wiki/Test-driven_development) to speed up your development.
   * Try to create or update [`docs/requirements`](https://devdocs.jabref.org/requirements/) to include a description of the requirement stemming from the issue.

>      Update your code to link to the requirement.

Reviewer: Same thoughts as in https://github.com/JabRef/jabref/pull/13727/files#r2290555949 - needs to be discussed with maintainers first if a contributor is to do this.

8. Wait for automatic checks to run and bots commenting.
9. Address the feedback of the automated checks. To find solutions to the most common errors that lead to such failures, check our [FAQ page](https://devdocs.jabref.org/code-howtos/faq).
10. Wait for feedback of one of the maintainers. Since this is a hobby project for all of them, this may take a while. You can speed up things by reducing their load. You can, for instance, try out other pull requests, review other pull requests, and answer questions on JabRef in the forum.
11. Address the feedback of the maintainer. In case your pull request changed something significant, we might ask you to create or update [`docs/requirements`](https://devdocs.jabref.org/requirements/) to include a description of the requirement stemming from the issue - and link that with your implementation.
12. Wait for feedback of a second maintainer.
13. Address the feedback of the second maintainer.
14. After two maintainers gave their green flag, the pull request will be merged.

We view pull requests as a collaborative process.
Submit a pull request early to enable feedback from the team while you continue working.

> Please also remember to discuss bigger changes early with the core maintainers to ensure properly spend time and work.

Reviewer: ```suggestion
Please also remember to discuss bigger changes early with the core maintainers to ensure more fruitful investment of time and work.
```
